### PR TITLE
Reason text for sentience potion

### PR DESCRIPTION
**Pull request by ErdGinalD**
## What Does This PR Do
Added reason text pop-up when you try to give animal sentience. Now ghosts would know you wanna some new friends or do little silly massacre. 

## Why It's Good For The Game
I guess no one want to see x99 attempt to give some bees sentience without any explains. This PR make easier to find friend to chat, or arise sentience army, leting ghost know your intentions without "LOOC: STOP PLEASE A DONT WANNA BE FLYING HORN POTATO". Much easier to invite ghost for gimmicks. via "I want to turn diamond golem into Senator Armstrong for lulz", not just ordinary boring serve.

## Images of changes
Activating hidden imba creature..
image

.. and creature realize: HIS TIME HAS COME!
image

## Testing
Launch 2 player local, give dogo potion with pop-up text reason, ghosts see reason why dude wanna arise dogo ultramind.

## Changelog
:cl:
add: Added reason text pop-up for sentience potion
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/SSghost_spawns.dm [hunk 1]
```diff
@@ -132,7 +132,7 @@ SUBSYSTEM_DEF(ghost_spawns)
 		if(isatom(source))
 			act_jump = "<a href='?src=[M.UID()];jump=\ref[source]'>\[Teleport]</a>"
 		var/act_signup = "<a href='?src=[A.UID()];signup=1'>\[Sign Up]</a>"
-		to_chat(M, "<big><span class='boldnotice'>Now looking for candidates [role ? "to play as \an [role_cleanname || get_roletext(role)]" : "\"[question]\""]. [act_jump] [act_signup]</span></big>")
+		to_chat(M, "<big><span class='boldnotice'>Now looking for candidates [role ? "to play as \an [role_cleanname || get_roletext(role)]" : "\"[question]\""]. [act_jump] [act_signup] [reason?"<i>\nReason: [reason]</i>":""]</span></big>")
```
Comment (lewcc) [inline code/controllers/subsystem/SSghost_spawns.dm:135]: There might be a better sanitize option for this but this is probably enough?
```suggestion
		to_chat(M, "<big><span class='boldnotice'>Now looking for candidates [role ? "to play as \an [role_cleanname || get_roletext(role)]" : "\"[question]\""]. [act_jump] [act_signup] [reason?"<i>\nReason: [sanitize(reason)]</i>":""]</span></big>")
```

**Review by lewcc — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/SSghost_spawns.dm (hunk 1 above)
Comment (ErdGinalD) [reply code/controllers/subsystem/SSghost_spawns.dm]: what a difference? i didnt saw visual changes about it

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -228,11 +228,12 @@
 			explosion(T, -1, -1, 2, 3)
 		qdel(src)
 		return
+	var/reason_text = input(user, "Enter reason for giving sentience", "Reason for sentience potion", "") as null|text
 	to_chat(user, "<span class='notice'>You offer [src] sentience potion to [SM]...</span>")
 	being_used = TRUE
 
-	var/ghostmsg = "Play as [SM.name], pet of [user.name]?"
-	var/list/candidates = SSghost_spawns.poll_candidates(ghostmsg, ROLE_SENTIENT, FALSE, 10 SECONDS, source = M)
+	var/ghostmsg = "Play as [SM.name], pet of [user.name]?[reason_text? "\nReason: [reason_text]\n":""]"
```
Comment (lewcc) [inline code/modules/research/xenobiology/xenobiology.dm:235]:
```suggestion
	var/ghostmsg = "Play as [SM.name], pet of [user.name]?[reason_text ? "\nReason: [sanitize(reason_text)]\n" : ""]"
```

**Review by lewcc — COMMENTED** [review]
Oops, missed this one.

**Comment by Henri215** [comment]
image
Changelog is missing these "CL"

**Comment by ErdGinalD** [comment]
> image Changelog is missing these "CL"

Is that okay?

**Comment by Henri215** [comment]
> > image Changelog is missing these "CL"
> 
> Is that okay?

Only missing the "/" before the last "CL"

**Comment by ErdGinalD** [comment]
> > > image Changelog is missing these "CL"
> > 
> > 
> > Is that okay?
> 
> Only missing the "/" before the last "CL"

Done. My first PR here, thanks for the support with that

File: code/controllers/subsystem/SSghost_spawns.dm [hunk 3]
```diff
@@ -132,7 +132,7 @@ SUBSYSTEM_DEF(ghost_spawns)
 		if(isatom(source))
 			act_jump = "<a href='?src=[M.UID()];jump=\ref[source]'>\[Teleport]</a>"
 		var/act_signup = "<a href='?src=[A.UID()];signup=1'>\[Sign Up]</a>"
-		to_chat(M, "<big><span class='boldnotice'>Now looking for candidates [role ? "to play as \an [role_cleanname || get_roletext(role)]" : "\"[question]\""]. [act_jump] [act_signup]</span></big>")
+		to_chat(M, "<big><span class='boldnotice'>Now looking for candidates [role ? "to play as \an [role_cleanname || get_roletext(role)]" : "\"[question]\""]. [act_jump] [act_signup] [reason?"<i>\nReason: [sanitize(reason)]</i>":""]</span></big>")
```
Comment (lewcc) [inline code/controllers/subsystem/SSghost_spawns.dm:135]:
```suggestion
		to_chat(M, "<big><span class='boldnotice'>Now looking for candidates [role ? "to play as \an [role_cleanname || get_roletext(role)]" : "\"[question]\""]. [act_jump] [act_signup] [reason ? "<i>\nReason: [sanitize(reason)]</i>" : ""]</span></big>")
```

**Comment by SteelSlayer** [comment]
Author has linked their account.

**Review by S34NW — APPROVED** [review]

**Pull request merged**